internal/scheduler/database/job_repository.go

```diff
@@ -203,11 +211,10 @@ func (r *PostgresJobRepository) FetchJobRunErrors(ctx *armadacontext.Context, ru
 		FROM %s as tmp
 		JOIN job_run_errors ON job_run_errors.run_id = tmp.run_id`
 
-			rows, err := tx.Query(ctx, fmt.Sprintf(query, tmpTable))
-			if err != nil {
-				return err
+			rows, queryErr = tx.Query(ctx, fmt.Sprintf(query, tmpTable))
+			if queryErr != nil {
+				return queryErr
 			}
-			defer rows.Close()
```

Comment:
@dejanzele I don't think this should be an issue, and I don't think it leaks the memory.

The defer once evaluated pushes the arguments to the defer stack at the time defer statement is evaluated. In this case, the address of the object returned from the `tx.Query` will be pushed. Even if `rows` is re-assigned, the copy of the memory address is pushed to the stack, so once the defer gets executed, it would call close on the correct object.

Since this is already correct from the behavioral point of view, the only question is should we close the row right after we are done with it. If so, I'd argue that the logic should be extracted to a function that would do everything inside the function body. Otherwise, I don't think the fix is necessary.